src/Xamarin.Android.Build.Tasks/Tasks/Aapt2Link.cs

```diff
@@ -120,8 +119,21 @@ public async override System.Threading.Tasks.Task RunTaskAsync ()
 						}
 					}
 				}
-				if (!string.IsNullOrEmpty (ProguardRuleOutput))
-					Files.CopyIfChanged (proguardRuleOutputTemp, ProguardRuleOutput);
+				if (!string.IsNullOrEmpty (ProguardRuleOutput)) {
+					// combine the "proguard" temp files into one file.
+					var sb = new StringBuilder ();
+					HashSet<string> output = new HashSet<string> ();
```

Comment:
This should also be a `SortedSet<string>`, so that the contents of the generated output are consistent and repeatable across builds.

Reply:
Using a `SortedSet` would re-order the contents of the first fle. The issue with that is there are comments in the `aapt_rules.txt` which match up with the lines. e.g

```
# Referenced at /Users/xxx/Projects/foofoo/foofoo.Android/obj/Release/android/manifest/AndroidManifest.xml:17
-keep class android.app.Application { <init>(); }
-keep class androidx.core.app.CoreComponentFactory { <init>(); }
# Referenced at /Users/xxx/Projects/foofoo/foofoo.Android/obj/Release/android/manifest/AndroidManifest.xml:30
-keep class com.google.android.gms.common.api.GoogleApiActivity { <init>(); }
```

Reply:
Not sure if we should try to keep the ordering in this case.